request.js

```diff
@@ -459,6 +464,10 @@ TChannelRequest.prototype.shouldRetryError = function shouldRetryError(err) {
         return false;
     }
 
+    if (self.channel.currentRetryRatio() > self.maxRetryRatio) {
+        return false;
```

Comment:
I think emitting a new metric is the most straightforward way to see how this feature is performing